cub/test/internal/catch2_test_integer_utils.cu

```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+#include <cub/detail/integer_utils.cuh>
+
+#include "c2h/catch2_test_helper.h"
+
+/***********************************************************************************************************************
+ * TEST CASES
+ **********************************************************************************************************************/
+
+using integral_types =
+  c2h::type_list<int16_t,
+                 uint16_t,
+                 int32_t,
+                 uint32_t,
+                 int64_t,
+                 uint64_t
+#if TEST_INT128()
+                 ,
+                 __int128_t,
+                 __uint128_t
+#endif
+                 >;
+
+using floating_point_types =
+  c2h::type_list<float,
+                 double
+#if TEST_HALF_T()
+                 ,
+                 __half
+#endif
+#if TEST_BF_T()
+                 ,
+                 __nv_bfloat16
+#endif
+                 >;
+
+using operator_types = c2h::type_list<cuda::minimum<>, cuda::maximum<>>;
+
+template <typename T>
+static __global__ void test_int_kernel(const T* input, int num_items)
+{
+  auto i = threadIdx.x + blockIdx.x * blockDim.x;
+  if (i < num_items)
+  {
+    auto value       = input[i];
+    auto [high, low] = cub::detail::split_integers(value);
+    auto result      = cub::detail::merge_integers(high, low);
+    assert(value == result);
```

Comment:
Suggestion: add an assertion on the values in-between splitting and merging. Also add a static assertion on the types of `high` and `low`.